src/Entity/Agreement.php

```diff
@@ -208,6 +208,21 @@ public function postSave(EntityStorageInterface $storage, $update = TRUE) {
     }
   }
 
+  // Delete also config entity associated with the Agreement
```

Comment:
I had to add hook for deleting the kifi_config entities. Since I noticed during development, they didn't get deleted when the agreement that was associated with was.